ERCS/erc-7573.md

```diff
@@ -0,0 +1,102 @@
+---
+eip: 7573
+title: Conditional-upon-Transfer-Decryption for DvP
+description: A Proposal for a Lean and Functional Delivery versus Payment
+author: Christian Fries (@cfries), Peter Kohl-Landgraf (@pekola)
+discussions-to: https://ethereum-magicians.org/t/erc-7573-conditional-upon-transfer-decryption-for-delivery-versus-payment/17232
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-12-05
+---
+
+## Abstract
+
+The interfaces model the functional transaction scheme to establish a secure delivery-versus-payment
+across two blockchains, where a) no intermediary is required and b) one of the two chains
+(e.g. the "payment chain") can securely interact with a stateless "decryption oracle".
+
+For convenience, we call the two chains "asset chain" and "payment chain". However, the concept is agnostic to the nature of the tokens managed on the two chains.
+
+The scheme is realized by two smart contracts, one on each chain.
+One smart contract implements the `ILockingContract` interface on one chain (e.g. the "asset chain"), and another smart contract implements the `IDecryptionContract` interface on the other chain (e.g., the "payment chain").
+
+The smart contract on the asset chain, implementing `ILockingContract`, locks the asset until a key is presented that will encrypt to a given value.
+
+The smart contract on the payment chain, implementing `IDecryptionContract`, decrypts one of two keys (via the decryption oracle) conditional to the success or failure of the token transfer initiated
+by the method
+```solidity
+function transferAndDecrypt(bytes32 id, address to, string memory keyEncryptedSuccess, string memory keyEncryptedFailure) external;
+```
+
+The contract  on the asset chain, implementing `ILockingContract`, will lock the token until a key is presented
+releasing the token by calling
+```solidity
+function transferWithKey(bytes32 id, string memory key) external;
+```
+where `key` is the decrypted `keyEncryptedSuccess` or `keyEncryptedFailure`
+
+For the decryption, a stateless decryption oracle is attached to the chain running `IDecryptionContract`.
+
+## Motivation
+
+Within the domain of financial transactions and distributed ledger technology (DLT), the Hash-Linked Contract (HLC) concept has been recognized as valuable and has been thoroughly investigated.
+The concept may help to solve the challenge of delivery-versus-payment (DvP), especially in cases where the asset chain and payment system (which may be a chain, too) are separated. The proposed solutions are based on an API-based interaction mechanism which bridges the communication between a so-called Asset Chain and a corresponding Payment System or requires complex and problematic time-locks (\cite{BancaItalia}). We believe an even more lightweight interaction across both systems is possible, especially when the payment system is also based on a DLT infrastructure.
```

Comment:
This is good background, but I think you need to expand on "why does the ecosystem need Conditional-upon-Transfer-Decryption for Delivery versus Payment?"

This section should answer the questions: who is going to be implementing this proposal, what use cases are enabled by it, and how does this particular design improve on prior art (if any exists)?